src/map/utils/zoneutils.cpp

```diff
@@ -500,7 +500,9 @@ void LoadMOBList()
                 PMob->m_TrueDetection = Sql_GetUIntData(SqlHandle, 65);
                 PMob->m_Detects = Sql_GetUIntData(SqlHandle, 66);
 
-                PMob->setMobMod(MOBMOD_CHARMABLE, Sql_GetUIntData(SqlHandle, 67));
+                if (GetZone(ZoneID)->GetType() != ZONETYPE_DYNAMIS && GetZone(ZoneID)->GetType() != ZONETYPE_BATTLEFIELD &&
```

Comment:
I think these checks should be done inside `mobutils::calculatestats`. There you can change the charmable value just before the mob spawns. It's just easier to debug when all these checks are done in one place.